packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -187,9 +188,25 @@ export class ModularChangeFamily
 			return makeModularChangeset();
 		}
 
-		return changes
-			.map((change) => change.change)
-			.reduce((change1, change2) => this.composePair(change1, change2, revInfos, idState));
+		const innerChanges = changes.map((change) => change.change);
+
+		// Recursively compose the left and right halves of the changeset list before composing their respective compositions.
+		// This leads to the same number of compositions as when composing the changesets in order,
+		// but in scenarios where the changesets grow as they are composed
+		// (e.g., when composing changesets that target different elements of a sequence),
+		// this approach reduces the number of times the individual change atoms within the changesets are processed.
+		// It benefits from the same principle that makes merge sort faster than insertion sort,
```

Comment:
It's obvious given the diff, but I think this comment could benefit with an approach highlighting the alternative here. e.g.

```suggestion
		// When compared with an approach like composing all the changesets left-to-right, this benefits from the same principle that makes merge sort faster than insertion sort,
```

(maybe make the change yourself to keep line length reasonable)